src/coreComponents/python/modules/geosx_mesh_doctor/checks/reorient_mesh.py

```diff
@@ -0,0 +1,171 @@
+import logging
+from typing import List, Tuple, Dict, FrozenSet, Iterator
+import numpy
+
+from tqdm import tqdm
+
+import networkx
+
+from vtkmodules.vtkCommonCore import (
+    vtkIdList,
+    vtkPoints,
+)
+from vtkmodules.vtkCommonDataModel import (
+    VTK_POLYHEDRON,
+    VTK_TRIANGLE,
+    vtkCellArray,
+    vtkPolyData,
+    vtkPolygon,
+    vtkUnstructuredGrid,
+    vtkTetra,
+)
+from vtkmodules.vtkFiltersCore import (
+    vtkTriangleFilter,
+)
+from .vtk_utils import (
+    to_vtk_id_list,
+)
+
+from .vtk_polyhedron import (
+    FaceStream,
+    build_cell_graph,
+)
+
+
+def __compute_volume(mesh_points: vtkPoints, face_stream: FaceStream) -> float:
```

Comment:
Regarding this issue: https://github.com/GEOSX/GEOSX/issues/2253
Are you using the "right" way to compute the cell volume? Does it matter here?

Reply:
Actually I do the computation my self (not using any `vtk` algo) and no, it shouldn't be a problem.
What matters is not really the value; but the fact that one is positive and the other negative (and even exactly opposite).
If we happen to have normals directed outwards with a negative volume (or inwards with a positive volume), then most probably the considered polyhedron is strongly self-intersecting and we'll detect this with other checks.